### ISSUE TITLE
Exception thrown when whole dataset fails validation using Polars

### ISSUE THREAD

**Issue by GeorgelPreput** [canimus/cuallee#106]
**Describe the bug**
When using Polars, if the given dataset fails validation, instead of getting a dataframe with the validation results, an exception is thrown: `TypeError: '>=' not supported between instances of 'NoneType' and 'float'`

The example below shows the use of the `is_unique` check, but this has been replicated with other checks as well.

**To Reproduce**
1. Create a new Jupyter notebook / Python file
2. Paste the following lines in:
```python
import polars as pl
from cuallee import Check, CheckLevel

df = pl.DataFrame(
    {
        "id": [1, 1, 2, 3, 4],
        "bar": [6, 7, 8, 9, 10],
        "ham": ["a", "b", "c", "d", "e"],
    }
)

id_check = Check(CheckLevel.WARNING, "ID unique")
display(id_check.is_unique("id").validate(df))
```

**Expected behavior**
Code returns a dataframe with validation results.

**Actual behavior**
Code throws exception below:

> ---------------------------------------------------------------------------
> TypeError                                 Traceback (most recent call last)
> Cell In[28], line 13
>       4 df = pl.DataFrame(
>       5     {
>       6         "id": [1, 1, 2, 3, 4],
>    (...)
>       9     }
>      10 )
>      12 id_check = Check(CheckLevel.WARNING, "ID unique")
> ---> 13 display(id_check.is_unique("id").validate(df))
> 
> File ~/Source/replicate-issue/.venv/lib/python3.11/site-packages/cuallee/__init__.py:586, in Check.validate(self, dataframe)
>     581     self.compute_engine = importlib.import_module("cuallee.polars_validation")
>     583 assert self.compute_engine.validate_data_types(
>     584     self.rules, dataframe
>     585 ), "Invalid data types between rules and dataframe"
> --> 586 return self.compute_engine.summary(self, dataframe)
> 
> File ~/Source/replicate-issue/.venv/lib/python3.11/site-packages/cuallee/polars_validation.py:414, in summary(check, dataframe)
>     410     return "FAIL"
>     412 rows = len(dataframe)
> --> 414 computation_basis = [
>     415     {
>     416         "id": index,
>     417         "timestamp": check.date.strftime("%Y-%m-%d %H:%M:%S"),
>     418         "check": check.name,
>     419         "level": check.level.name,
>     420         "column": str(rule.column),
>     421         "rule": rule.method,
>     422         "value": rule.value,
>     423         "rows": rows,
>     424         "violations": _calculate_violations(first(unified_results[hash_key]), rows),
>     425         "pass_rate": _calculate_pass_rate(first(unified_results[hash_key]), rows),
>     426         "pass_threshold": rule.coverage,
>     427         "status": _evaluate_status(
>     428             _calculate_pass_rate(first(unified_results[hash_key]), rows),
>     429             rule.coverage,
>     430         ),
>     431     }
>     432     for index, (hash_key, rule) in enumerate(check._rule.items(), 1)
>     433 ]
>     434 pl.Config.set_tbl_cols(12)
>     435 return pl.DataFrame(computation_basis)
> 
> File ~/Source/replicate-issue/.venv/lib/python3.11/site-packages/cuallee/polars_validation.py:427, in (.0)
>     410     return "FAIL"
>     412 rows = len(dataframe)
>     414 computation_basis = [
>     415     {
>     416         "id": index,
>     417         "timestamp": check.date.strftime("%Y-%m-%d %H:%M:%S"),
>     418         "check": check.name,
>     419         "level": check.level.name,
>     420         "column": str(rule.column),
>     421         "rule": rule.method,
>     422         "value": rule.value,
>     423         "rows": rows,
>     424         "violations": _calculate_violations(first(unified_results[hash_key]), rows),
>     425         "pass_rate": _calculate_pass_rate(first(unified_results[hash_key]), rows),
>     426         "pass_threshold": rule.coverage,
> --> 427         "status": _evaluate_status(
>     428             _calculate_pass_rate(first(unified_results[hash_key]), rows),
>     429             rule.coverage,
>     430         ),
>     431     }
>     432     for index, (hash_key, rule) in enumerate(check._rule.items(), 1)
>     433 ]
>     434 pl.Config.set_tbl_cols(12)
>     435 return pl.DataFrame(computation_basis)
> 
> File ~/Source/replicate-issue/.venv/lib/python3.11/site-packages/cuallee/polars_validation.py:407, in summary.._evaluate_status(pass_rate, pass_threshold)
>     405 def _evaluate_status(pass_rate, pass_threshold):
> --> 407     if pass_rate >= pass_threshold:
>     408         return "PASS"
>     410     return "FAIL"
> 
> TypeError: '>=' not supported between instances of 'NoneType' and 'float'

**Desktop (please complete the following information):**
 - OS: Linux
 - cuallee = "^0.4.5"
 - ipykernel = "^6.24.0"
 - polars = "0.18.7"

**Additional context**
None.

**Comment by canimus** [comment]
Thanks for reporting the bug @GeorgelPreput.
Polars is changing rapidly, and the methods to access the first row of a DataFrame have changed since the earlier versions. That was causing the issue. Should be fixed in `cuallee==0.4.7` which is published in the python index already.

**Comment by GeorgelPreput** [comment]
Thanks for fixing it so fast! :slightly_smiling_face:
